docs/concepts/decorators-and-auto-logging.md

```diff
@@ -87,6 +87,8 @@ def train_and_predict(train_data: pd.DataFrame,
 ...
 ```
 
+**Notice**: `typing` hints like `Optional`, `Union` and `List` are currently not supported but will be in the future. 
```

Comment:
I think a message like the following will make it clearer that the whole typing library is not supported
```suggestion
**Notice**: type hints from the`typing` library (e.g. `typing.Optional`, `typing.Union`, `typing.List` etc.) are currently not supported but will be in the future. 
```